docs/fundamentals/networking/websockets.md

```diff
@@ -99,3 +99,76 @@ using ClientWebSocket ws = new()
 
 > [!IMPORTANT]
 > Before using compression, please be aware that enabling it makes the application subject to CRIME/BREACH type of attacks, for more information, see [CRIME](https://en.wikipedia.org/wiki/CRIME) and [BREACH](https://en.wikipedia.org/wiki/BREACH). It is strongly advised to turn off compression when sending data containing secrets by specifying the `DisableCompression` flag for such messages.
+
+## Keep-Alive strategies
+
+On **.NET 8** and earlier, the only available Keep-Alive strategy is _Unsolicited PONG_. This strategy is enough to keep the underlying TCP connection from idling out. However, in a case when a remote host becomes unresponsive (for example, a remote server crashes), the only way to detect such situations with Unsolicited PONG is to depend on the TCP timeout.
+
+**.NET 9** introduced the long-desired _PING/PONG_ Keep-Alive strategy, complementing the existing `KeepAliveInterval` setting with the new `KeepAliveTimeout` setting. Starting with .NET 9, the Keep-Alive strategy is selected as follows:
+
+1. Keep-Alive is **OFF**, if
+    - `KeepAliveInterval` is `TimeSpan.Zero` or `Timeout.InfiniteTimeSpan`
+2. **Unsolicited PONG**, if
+    - `KeepAliveInterval` is a positive finite `TimeSpan`, -AND-
+    - `KeepAliveTimeout` is `TimeSpan.Zero` or `Timeout.InfiniteTimeSpan`
+3. **PING/PONG**, if
+    - `KeepAliveInterval` is a positive finite `TimeSpan`, -AND-
+    - `KeepAliveTimeout` is a positive finite `TimeSpan`
+
+The default `KeepAliveTimeout` value is `Timeout.InfiniteTimeSpan`, so the default Keep-Alive behavior remains consistent between .NET versions.
+
+If you use `ClientWebSocket`, the default <xref:System.Net.WebSockets.ClientWebSocketOptions.KeepAliveInterval?displayProperty=nameWithType> value is <xref:System.Net.WebSockets.WebSocket.DefaultKeepAliveInterval?displayProperty=nameWithType> (typically 30 seconds). That means, `ClientWebSocket` has the Keep-Alive ON by default, with Unsolicited PONG as the default strategy.
+
+If you want to switch to the PING/PONG strategy, overriding <xref:System.Net.WebSockets.ClientWebSocketOptions.KeepAliveTimeout?displayProperty=nameWithType> is enough:
```

Comment:
Would be good suggest that this is an implication of the behavior described before.

```suggestion
As a result, overriding <xref:System.Net.WebSockets.ClientWebSocketOptions.KeepAliveTimeout?displayProperty=nameWithType> is enough to switch to the PING/PONG strategy if you keep the default value for <xref:System.Net.WebSockets.WebSocket.DefaultKeepAliveInterval?displayProperty=nameWithType>:
```